dist/common/scripts/scylla_util.py

```diff
@@ -577,18 +624,37 @@ def is_gentoo_variant():
 def redhat_version():
     return distro.version()
 
-def is_ec2():
-    if os.path.exists('/sys/hypervisor/uuid'):
-        with open('/sys/hypervisor/uuid') as f:
-            s = f.read()
-        return True if re.match(r'^ec2.*', s, flags=re.IGNORECASE) else False
-    elif os.path.exists('/sys/class/dmi/id/board_vendor'):
-        with open('/sys/class/dmi/id/board_vendor') as f:
-            s = f.read()
-        return True if re.match(r'^Amazon EC2$', s) else False
+
+def get_text_from_path(fpath):
+    board_vendor_path = Path(fpath)
+    if board_vendor_path.exists():
+        return board_vendor_path.read_text().strip()
+    return ""
+
+def match_patterns_in_files(list_of_patterns_files):
+    for pattern, fpath in list_of_patterns_files:
+        if re.match(pattern, get_text_from_path(fpath), flags=re.IGNORECASE):
+            return True
     return False
 
 
+def is_ec2():
+    return aws_instance.is_aws_instance()
```

Comment:
also GCP actually just does dns query (but I'd love it to be optimized and use /sys or any other variable instead)

Reply:
@tarzanek That's how it was originally implemented but Avi was not happy about it:
comment from above:
![image](https://user-images.githubusercontent.com/5269241/99523124-ab96dc00-29a7-11eb-8184-1d5cf73f7dcf.png)

Reply:
so how will this method then behave on non cloud servers? or it will just wait for timeout?

Reply:
in the current implementation, it will wait for a timeout... but it's not critical since those functions are mainly used in the Scylla Amazon/GCE images